### PR TITLE
Update Documentation and a few fixes in HTTP

### PR DESCRIPTION
**Pull request by csanders-git**
Updated the YAML Format documentation

Also fixed HTTP post requests
Added support for Windows Blocking
Fixed unicode headers
Added stop_magic

### CODE REVIEW

**Comment by csanders-git** [comment]
Fixes #57

File: docs/YAMLFormat.md [hunk 1]
```diff
@@ -0,0 +1,414 @@
+===============
+The YAML Format
+===============
+
+Welcome to the the FTW YAMLFormat docuemntation. In this document we will explain all the possible options that can be used within the YAML format. Generally this is the prefereed format for writing tests in as they don't require any programming skills in order to understand and change. If you find a bug in this format please open an issue.
```
Comment (zmallen) [inline docs/YAMLFormat.md]: spelling errors here

File: ftw/http.py [hunk 2]
```diff
@@ -334,13 +335,13 @@ def build_request(self):
         headers = ''
         if self.request_object.headers != {}:
             for hname, hvalue in self.request_object.headers.iteritems():
-                headers += str(hname) + ': ' + str(hvalue) + str(self.CRLF)
+                headers += unicode(hname) + ': ' + unicode(hvalue) + str(self.CRLF)
```
Comment (zmallen) [inline ftw/http.py]: Can you give some insight about this fix?

File: ftw/http.py [hunk 3]
```diff
@@ -334,13 +335,13 @@ def build_request(self):
...
         request = string.replace(request, '#headers#', headers)
 
         # If we have data append it
         if self.request_object.data != '':
             data = str(self.request_object.data)
-            request = string.replace(request, '#data#', data)
+            request = string.replace(request, '#data#', data + self.CRLF)
```
Comment (zmallen) [inline ftw/http.py]: Can you give some insight about this fix?

File: ftw/http.py [hunk 4]
```diff
@@ -391,6 +391,9 @@ def get_response(self):
                 if err.errno == errno.EAGAIN:
                     pass
                 # SSL will return SSLWantRead instead of EAGAIN
+                elif sys.platform == 'win32' and \
```
Comment (zmallen) [inline ftw/http.py]: Can you give some insight about this fix?

File: docs/YAMLFormat.md (hunk 1 above)
Comment (csanders-git) [inline docs/YAMLFormat.md]: I accept PRs :-P

Comment (csanders-git) [inline docs/YAMLFormat.md]: Or more specific instructions as to where :-P

File: ftw/ruleset.py [hunk 5]
```diff
@@ -8,11 +8,12 @@ class Output(object):
     """
     This class holds the expected output from a corresponding FTW HTTP Input
     We are stricter in this definition by requiring at least one of status,
-    html_contains or log_contains
+    html_contains,no_log_contains, or log_contains
     """
     def __init__(self, output_dict):
         self.STATUS = 'status'
         self.LOG = 'log_contains'
+        self.NOTLOG = 'no_log_contains'
```
Comment (zmallen) [inline ftw/ruleset.py]: Why do we specify no_log_contains?

File: docs/YAMLFormat.md [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+
+**Example Usage:** ```log_contains: "id:1234"```
+
+**Default Value:** None
+
+**Scope:** Output
+
+**Version:** 0.1
+
+
```
Comment (zmallen) [inline docs/YAMLFormat.md]: Can you add `no_log_contains` in this documentation?

Comment (csanders-git) [inline docs/YAMLFormat.md]: already added

File: ftw/ruleset.py (hunk 5 above)
Comment (csanders-git) [inline ftw/ruleset.py]: what's your question?

File: ftw/http.py (hunk 4 above)
Comment (csanders-git) [inline ftw/http.py]: gladly it'd be easier offline

File: ftw/http.py (hunk 3 above)
Comment (csanders-git) [inline ftw/http.py]: HTTP requires a trailing /r/n which was manually added before but it's nicer if it works like magic.

**Comment by zmallen** [comment]
YO TRAV HIT THAT BEAT

**Comment by csanders-git** [comment]
@zmallen you're a nerd.

**Pull request merged**